packages/scandipwa/src/component/CheckoutShipping/CheckoutShipping.container.js

```diff
@@ -105,27 +106,29 @@ export class CheckoutShippingContainer extends PureComponent {
         };
     }
 
-    componentDidUpdate(prevProps) {
-        const { shippingMethods: prevShippingMethods } = prevProps;
-        const { shippingMethods } = this.props;
+    componentDidMount() {
+        const { isPickInStoreMethodSelected, shippingMethods } = this.props;
+        const instore = shippingMethods.find((el) => el.method_code === STORE_IN_PICK_UP_METHOD_CODE);
 
-        if (prevShippingMethods !== shippingMethods) {
-            this.resetShippingMethod();
+        if (isPickInStoreMethodSelected) {
+            this.onShippingMethodSelect(instore);
         }
     }
 
-    resetShippingMethod() {
-        const { selectedShippingMethod: { method_code: selectedMethodCode = '' } } = this.state;
-        const { shippingMethods } = this.props;
-
-        if (shippingMethods.find(({ method_code }) => method_code === selectedMethodCode)) {
-            return;
-        }
+    componentDidUpdate(prevProps) {
+        const { isPickInStoreMethodSelected, shippingMethods = [] } = this.props;
+        const { isPickInStoreMethodSelected: prevIsPickInStoreMethodSelected } = prevProps;
 
-        const [defaultShippingMethod] = shippingMethods.filter((method) => method.available);
-        const selectedShippingMethod = defaultShippingMethod || {};
+        const instore = shippingMethods.find((el) => el.method_code === STORE_IN_PICK_UP_METHOD_CODE);
```

Comment:
This code is repeated with the code in the `componentDidMount`. Please move this to a separate method and reuse it.

Reply:
done